cmd/snap/cmd_run.go

```diff
@@ -948,14 +948,21 @@ func (x *cmdRun) runCmdUnderStrace(origCmd []string, envForExec envForExecFunc)
 	// run with filter
 	cmd.Env = envForExec(nil)
 	cmd.Stdin = Stdin
-	cmd.Stdout = Stdout
+	// note hijacking stdout, means it is no longer a tty and programs
+	// expecting stdout to be on a terminal (eg. bash) may misbehave at this
+	// point
```

Comment:
I think we should print a warning to stderr to make it clear that using --strace has this behavior, as it is a change from previous years explicitly meant to avoid a bug.